sqlglot/expressions.py

```diff
@@ -2556,9 +2556,20 @@ def with_(
     "sample": False,
     "settings": False,
     "format": False,
+    "option": False,
 }
 
 
+# https://learn.microsoft.com/en-us/sql/t-sql/queries/option-clause-transact-sql?view=sql-server-ver16
+# https://learn.microsoft.com/en-us/sql/t-sql/queries/hints-transact-sql-query?view=sql-server-ver16
+class Option(Expression):
+    arg_types = {"option_type": True, "option_value": False, "requires_equals": False}
+
+
+class Options(Expression):
+    arg_types = {"options": False}
```

Comment:
Let's get rid of this and just store a list of `Option` in an `"options"` query modifier arg instead.

Reply:
Will do